### PR TITLE
Embrace `Etc/Unknown` timezone

### PR DESCRIPTION
**Pull request by robertbastian**
#5653

### CODE REVIEW

**Comment by sffc** [comment]
(we don't have consensus on this)

**Comment by robertbastian** [comment]
(that's why it's draft)

File: components/timezone/src/ids.rs [hunk 1]
```diff
@@ -209,27 +211,30 @@ impl TimeZoneIdMapperBorrowed<'_> {
...
+    pub fn normalize_iana<'s>(&self, iana_id: &'s str) -> (Cow<'s, str>, TimeZoneBcp47Id) {
+        self.iana_lookup_with_normalization(iana_id, |_| {})
+            .and_then(|(trie_value, string)| {
+                let Some(bcp47_id) = self.data.bcp47_ids.get(trie_value.index()) else {
+                    debug_assert!(false, "index should be in range");
+                    return None;
+                };
+                Some((string, bcp47_id))
+            })
+            .unwrap_or((Cow::Borrowed("Etc/Unknown"), TimeZoneBcp47Id::unknown()))
```
Comment (sffc) [inline components/timezone/src/ids.rs:237]: Issue: This function should return `None` if the time zone is not found. It should never map it to `Etc/Unknown`. Canonicalization can do that, but not normalization, please.

File: components/timezone/src/ids.rs [hunk 2]
```diff
@@ -248,37 +253,38 @@ impl TimeZoneIdMapperBorrowed<'_> {
...
     ///
     /// // Borrows when able:
-    /// let result = mapper.canonicalize_iana("America/Chicago").unwrap();
+    /// let result = mapper.canonicalize_iana("America/Chicago");
     /// assert_eq!(result.0, "America/Chicago");
     /// assert!(matches!(result.0, Cow::Borrowed(_)));
     ///
     /// // Unknown IANA time zone ID:
-    /// assert_eq!(mapper.canonicalize_iana("America/San_Francisco"), None);
+    /// assert_eq!(mapper.canonicalize_iana("America/San_Francisco").1, TimeZoneBcp47Id::unknown());
```
Comment (sffc) [inline components/timezone/src/ids.rs:268]: Nit: Also show the equivalence to "Etc/Unknown"

**Review by sffc — CHANGES_REQUESTED** [review]
I'm not super happy with descending from the type system to Etc/Unknown in the ID mapper. We definitely should not do it in the normalize function. In the canonicalize function, I can see an argument that we want it to be more transparent with the TimeZoneInfo struct.

You've sold me on _most_ of your time zone design things, but I still haven't heard a compelling argument about, what's so wrong with `None` and `Some(Etc/Unknown)` co-existing with slightly different meanings?

File: components/timezone/src/ids.rs (hunk 2 above)
Comment (robertbastian) [reply components/timezone/src/ids.rs]: I've made this fallible again, because `Etc/Unknown` is not a canonical zone (it's not an IANA zone at all).

**Review by sffc — APPROVED** [review]

**Pull request merged**